app/models/report.rb

```diff
@@ -48,10 +48,8 @@ def save_report_and_mentioning_reports
   private
 
   def save_mentioning_reports
-    content.scan(REPORT_LINK).uniq.each do |report_id|
-      Report.where(id: report_id).find_each do |report|
-        mentioning_reports << report
-      end
+    Report.where(id: content.scan(REPORT_LINK).uniq.flatten).find_each do |report|
```

Comment:
```suggestion
    report_ids = content.scan(REPORT_LINK).uniq.flatten
    Report.where(id: report_ids).find_each do |report|
```

一度変数に入れた方が可読性が高いかもしれません